test/e2e/plain_provisioner_test.go

```diff
@@ -1029,6 +1030,158 @@ var _ = Describe("plain provisioner bundleinstance", func() {
+			By("deleting the test Bundle resource")
+			Expect(c.Delete(ctx, b)).To(BeNil())
+
+			By("waiting until all the configmaps have been deleted")
+			selector := util.NewBundleLabelSelector(b)
+			Eventually(func() bool {
+				pods := &corev1.PodList{}
+				if err := c.List(ctx, pods, &client.ListOptions{
+					Namespace:     defaultSystemNamespace,
+					LabelSelector: selector,
+				}); err != nil {
+					return false
+				}
+				return len(pods.Items) == 0
+			}).Should(BeTrue())
+		})
+		It("should result in the underlying metadata/object configmaps being deleted", func() {
+			By("deleting the test Bundle resource")
+			Expect(c.Delete(ctx, b)).To(BeNil())
+
+			By("waiting until all the configmaps have been deleted")
+			selector := util.NewBundleLabelSelector(b)
+			Eventually(func() bool {
+				cms := &corev1.ConfigMapList{}
+				if err := c.List(ctx, cms, &client.ListOptions{
+					Namespace:     defaultSystemNamespace,
+					LabelSelector: selector,
+				}); err != nil {
+					return false
+				}
+				return len(cms.Items) == 0
+			}).Should(BeTrue())
+		})
+	})
+
+	When("a BundleInstance has been deleted", func() {
+		var (
+			ctx context.Context
+			b   *rukpakv1alpha1.Bundle
+			bi  *rukpakv1alpha1.BundleInstance
+		)
+		BeforeEach(func() {
+			ctx = context.Background()
+
+			By("creating the testing Bundle resource")
+			b = &rukpakv1alpha1.Bundle{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "e2e-ownerref-bundle-valid",
+				},
+				Spec: rukpakv1alpha1.BundleSpec{
+					ProvisionerClassName: plainProvisionerID,
+					Source: rukpakv1alpha1.BundleSource{
+						Type: rukpakv1alpha1.SourceTypeImage,
+						Image: &rukpakv1alpha1.ImageSource{
+							Ref: "testdata/bundles/plain-v0:valid",
+						},
+					},
+				},
+			}
+			Expect(c.Create(ctx, b)).To(BeNil())
+
+			By("creating the testing BI resource")
+			bi = &rukpakv1alpha1.BundleInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "e2e-ownerref-bi-valid",
+				},
+				Spec: rukpakv1alpha1.BundleInstanceSpec{
+					ProvisionerClassName: plainProvisionerID,
+					BundleName:           b.GetName(),
+				},
+			}
+			Expect(c.Create(ctx, bi)).To(BeNil())
+
+			By("waiting for the BI to eventually report a successful install status")
+			Eventually(func() (*metav1.Condition, error) {
+				if err := c.Get(ctx, client.ObjectKeyFromObject(bi), bi); err != nil {
+					return nil, err
+				}
+				return meta.FindStatusCondition(bi.Status.Conditions, rukpakv1alpha1.TypeInstalled), nil
+			}).Should(And(
+				Not(BeNil()),
+				WithTransform(func(c *metav1.Condition) string { return c.Type }, Equal(rukpakv1alpha1.TypeInstalled)),
+				WithTransform(func(c *metav1.Condition) metav1.ConditionStatus { return c.Status }, Equal(metav1.ConditionTrue)),
+				WithTransform(func(c *metav1.Condition) string { return c.Reason }, Equal(rukpakv1alpha1.ReasonInstallationSucceeded)),
+				WithTransform(func(c *metav1.Condition) string { return c.Message }, Equal("")),
```

Comment:
nit: do we need this message check if we don't set a message on a successful BI install?

Reply:
Heh IIRC Joe had a nit comment in one of my previous e2e PRs that asked for this message comparison despite the message being empty. I don't have a strong opinion but I'd lean towards whatever is already consistent throughout the codebase.

Reply:
My reasoning for that was to ensure we don't leave a stale message from a previous reconcile run. This says "we specifically expect message to be empty".